### DISCUSSION TITLE
Authorizing against a private mirror

### DISCUSSION THREAD

**Discussion by Splamy** [bagetter/BaGetter#88] [Ideas]
Hey,
I just recently stumbled across this server and so far really like it.

It seems like there's currently no possibility to set authorization for the upstream server or generally in the BaGetter.Protocol sdk.

Would this be a feature worthwhile adding?
If yes, I could try to contribute an implementation.

**Comment by FroggieFrog** [comment]
As stated in the docs, there is currently no support for Authorization.
https://www.bagetter.com/docs/configuration#private-feeds

**Reply by seriouz** [reply]
We started planning this feature in Discord.
Authorization also enables the possibility of private packages.

**Reply by Splamy** [reply]
Oh, maybe I misunderstood the documentation here, but I don't mean authentication for the bagetter feed, but for the upstream mirror. Something like
```jsonc
{
    ...
    "Mirror": {
        "Enabled":  true,
        "PackageSource": "https://nuget.pkg.github.com/Splamy/index.json",
        "Auth": { "User": "Splamy", "Pass": "asdf" }
    },
    ...
}
```

**Reply by seriouz** [reply]
Oh I think I understood now.
When I restore my NuGets through BaGetter and BaGetter mirrors a private GitHub registry, we have currently no possibility to authenticate against GitHub from BaGetter.

**Reply by Splamy** [reply]
Would you accept a PR for that? I've so far prototyped a simple implementation but could clean it up and add docu.

**Reply by seriouz** [reply]
Surely, we would accept a PR with this useful feature.
But give some thoughts about the possibility that you can authenticate with user+pw or sometimes an authtoken.
Docs about such feature should be mandatory. 
Thanks!
